tasks/fh-build.js

```diff
@@ -63,13 +64,17 @@ module.exports = function(grunt) {
       '!**/*.tar.gz'
     ];
     if (bundle_deps) {
-      _.map(_.keys(grunt.file.readJSON('package.json').dependencies),
-            function(dep) {
-              patterns.push('node_modules/' + dep + '/**');
-              return dep;
-            });
+      exec('npm ls --parseable --production').toString().split('\n').forEach(function(path) {
+        var pathParts = path.split('/');
+        var nodeModules = pathParts.indexOf('node_modules');
+        if(nodeModules >= 0 && nodeModules + 1 < pathParts.length) {
+          var parsedPath = [pathParts[nodeModules], pathParts[nodeModules + 1]].join('/') + '/**';
```

Comment:
A sample snippet from `npm ls --parseable --production`:
```
/home/tle/work/fh-ngui/node_modules/watchify
/home/tle/work/fh-ngui/node_modules/watchify/node_modules/assert
/home/tle/work/fh-ngui/node_modules/watchify/node_modules/base64-js
/home/tle/work/fh-ngui/node_modules/watchify/node_modules/browser-pack
```
`node_modules/watchify/**` is enough to include other sub dependencies